File: 20221026/모범답안/20221026_04/Palindrome.py
def solution(n):
    '''
    :param n: int
    :return: int
    '''

    if n == 1:
        return 10

    ans = 1

    for i in range(n // 2):
        ans = (ans * (9 if i == 0 else 10))
        ans %= 1_000_000_007
    
    if n % 2 == 1:
        ans = (ans * 10) % 1_000_000_007
    
    return ans

File: 20221026/모범답안/20221026_04/Palindrome.js

/**
 * 
 * @param n {number}
 * @return {number}
 */
function solution(n) {
    if (n == 1) {
        return 10;
    }

    let ans = 1;

    for (let i = 0; i < n; i += 2) {
         ans = (ans * (i == 0 ? 9 : 10));
         ans %= 1_000_000_007
    }

    return ans;
}